src/Microsoft.ML.LightGbm/LightGbmBinaryTrainer.cs

```diff
@@ -93,9 +96,79 @@ public sealed class LightGbmBinaryClassificationTrainer : LightGbmTrainerBase<fl
+            /// </summary>
+            /// <value>
+            /// This is useful for training on unbalanced data. A typical value to consider is sum(negative cases) / sum(positive cases).
+            /// </value>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Control the balance of positive and negative weights, useful for unbalanced classes." +
+                " A typical value to consider: sum(negative cases) / sum(positive cases).",
+                ShortName = "ScalePosWeight")]
+            public double WeightOfPositiveExamples = 1;
+
+            /// <summary>
+            /// Parameter for the sigmoid function.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Parameter for the sigmoid function.", ShortName = "sigmoid")]
+            [TGUI(Label = "Sigmoid", SuggestedSweeps = "0.5,1")]
+            public double Sigmoid = 0.5;
+
+            /// <summary>
+            /// Determines what evaluation metric to use.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Evaluation metrics.",
+                ShortName = "em")]
+            public EvaluateMetricType EvaluationMetric = EvaluateMetricType.Logloss;
+
+            static Options()
+            {
+                NameMapping.Add(nameof(EvaluateMetricType),                 "metric");
+                NameMapping.Add(nameof(EvaluateMetricType.None),            "");
+                NameMapping.Add(nameof(EvaluateMetricType.Logloss),         "binary_logloss");
+                NameMapping.Add(nameof(EvaluateMetricType.Error),           "binary_error");
+                NameMapping.Add(nameof(EvaluateMetricType.AreaUnderCurve),  "auc");
+                NameMapping.Add(nameof(WeightOfPositiveExamples),           "scale_pos_weight");
+            }
+
+            internal override Dictionary<string, object> ToDictionary(IHost host)
+            {
+                var res = base.ToDictionary(host);
+                res[GetOptionName(nameof(UnbalancedSets))] = UnbalancedSets;
+                res[GetOptionName(nameof(WeightOfPositiveExamples))] = WeightOfPositiveExamples;
+                res[GetOptionName(nameof(Sigmoid))] = Sigmoid;
+                if(EvaluationMetric != EvaluateMetricType.Default)
+                    res[GetOptionName(nameof(EvaluateMetricType))] = GetOptionName(EvaluationMetric.ToString());
+
+                return res;
+            }
+        }
+
         internal LightGbmBinaryClassificationTrainer(IHostEnvironment env, Options options)
              : base(env, LoadNameValue, options, TrainerUtils.MakeBoolScalarLabel(options.LabelColumnName))
         {
+            Contracts.CheckUserArg(options.Sigmoid > 0, nameof(Options.Sigmoid), "must be > 0.");
+            Contracts.CheckUserArg(options.WeightOfPositiveExamples > 0, nameof(Options.WeightOfPositiveExamples), "must be >= 0.");
```

Comment:
>>= [](start = 124, length = 2)

>

Reply:
nice catch

---
In reply to: [266671856](https://github.com/dotnet/machinelearning/pull/2948#discussion_r266671856) [](ancestors = 266671856)